dm/unit/unit.go

```diff
@@ -54,9 +57,20 @@ type Unit interface {
 
 // NewProcessError creates a new ProcessError
 // we can refine to add error scope field if needed
-func NewProcessError(errorType pb.ErrorType, msg string) *pb.ProcessError {
-	return &pb.ProcessError{
+func NewProcessError(errorType pb.ErrorType, err error) *pb.ProcessError {
+	result := &pb.ProcessError{
 		Type: errorType,
-		Msg:  msg,
+		Msg:  errors.ErrorStack(err),
```

Comment:
Shall we remove this `Msg` someday?

Reply:
Maybe not, because `error stack` is not easy to protobuf encoded, we have to include the error stack information in some field, either in the `Msg` or addition `Msg` field in `Error`. To keep the compatibility `Msg` field is still used.